src/control/controlpushbutton.h

```diff
@@ -27,42 +28,36 @@
 class ControlPushButton : public ControlObject {
     Q_OBJECT
   public:
-    enum ButtonMode {
-         PUSH = 0,
-         TOGGLE,
-         POWERWINDOW,
-         LONGPRESSLATCHING,
-         TRIGGER,
-    };
-
-    static QString buttonModeToString(int mode) {
+    static QString buttonModeToString(ControlButtonMode mode) {
         switch(mode) {
-            case ControlPushButton::PUSH:
-                return "PUSH";
-            case ControlPushButton::TOGGLE:
-                return "TOGGLE";
-            case ControlPushButton::POWERWINDOW:
-                return "POWERWINDOW";
-            case ControlPushButton::LONGPRESSLATCHING:
-                return "LONGPRESSLATCHING";
-            case ControlPushButton::TRIGGER:
-                return "TRIGGER";
-            default:
-                return "UNKNOWN";
+        case ControlButtonMode::PUSH:
+            return "PUSH";
+        case ControlButtonMode::TOGGLE:
+            return "TOGGLE";
+        case ControlButtonMode::POWERWINDOW:
+            return "POWERWINDOW";
+        case ControlButtonMode::LONGPRESSLATCHING:
+            return "LONGPRESSLATCHING";
+        case ControlButtonMode::TRIGGER:
+            return "TRIGGER";
+        default:
+            return "UNKNOWN";
```

Comment:
do these strings require stability? Can we change them? lets remove the default case because it should not be possible to pass an invalid value and when the enum gets changed, we get notified thanks to `-Wswitch`. Also making `ControlButtonMode` a Q_ENUM would make this redundant, wdyt?
```suggestion
        case ControlButtonMode::Push:
            return QStringLiteral("Push");
        case ControlButtonMode::Toggle:
            return QStringLiteral("Toggle");
        case ControlButtonMode::PowerWindow:
            return QStringLiteral("PowerWindow");
        case ControlButtonMode::LongPressLatching:
            return QStringLiteral("LongPressLatching");
        case ControlButtonMode::Trigger:
            return QStringLiteral("Trigger");
```

Reply:
Q_ENUM did not work here, because this is a static member function and Q_ENUM works only for membar variables.